cpp/inc/bond/ext/grpc/unary_call.h

```diff
@@ -35,61 +28,36 @@ class shared_unary_call;
 /// @note This class can only be moved. If shared ownership semantics are
 /// needed, convert this to a \ref shared_unary_call with the \ref share
 /// function.
-///
-/// @warning A moved-from unary_call is only valid for destruction or
-/// assignment.
-template <typename TRequest, typename TResponse>
-class unary_call final : public detail::unary_call_base<TRequest, TResponse>
+template <typename Request, typename Response>
+class unary_call final : public detail::unary_call_base<Request, Response>
 {
-    using base_type = detail::unary_call_base<TRequest, TResponse>;
+    using base_type = typename unary_call::unary_call_base;
 
 public:
-    /// @brief The default constructor of unary_call is not callable: the
-    /// Bond-over-gRPC infrastructure will create unary_call instances as
-    /// needed.
-    ///
-    /// Once created, a unary_call instance can only be moved to another
-    /// unary_call or converted to a \ref shared_unary_call (see the \ref
-    /// share function).
-    unary_call() = delete;
+    using base_type::base_type;
 
-    // unary_call is move-only
     unary_call(const unary_call&) = delete;
     unary_call& operator=(const unary_call&) = delete;
 
-    /// @brief Constructs a unary_call by moving from another unary_call.
-    ///
-    /// The moved-from instance is valid only for destruction and
-    /// assignment.
     unary_call(unary_call&& other) = default;
-
-    /// @brief Assign to this unary_call by moving from another unary_call.
-    ///
-    /// The moved-from instance is valid only for destruction and
-    /// assignment.
     unary_call& operator=(unary_call&& rhs) = default;
 
+    using base_type::operator bool;
```

Comment:
I think this pattern was to get the documentation for the functions to render in the non-detail types instead of just in the base when running Doxygen. We should check if that does what we think it does. If not, `BOND_DOXYGEN_ONLY` or leave this line off.